common/scala/src/main/scala/whisk/connector/kafka/KafkaMessagingProvider.scala

```diff
@@ -46,15 +57,14 @@ object KafkaMessagingProvider extends MessagingProvider {
   def getProducer(config: WhiskConfig, ec: ExecutionContext)(implicit logging: Logging): MessageProducer =
     new KafkaProducerConnector(config.kafkaHosts, ec)
 
-  def ensureTopic(config: WhiskConfig, topic: String, topicConfig: Map[String, String])(
-    implicit logging: Logging): Boolean = {
+  def ensureTopic(config: WhiskConfig, topic: String, topicConfig: String)(implicit logging: Logging): Boolean = {
```

Comment:
Is config still needed?

Reply:
I'm still using it to read kafka.replicationFactor.   I think we should move that value to kafka.conf as well, but I didn't want to delay getting this PR merged to trace through all the deployment configs that might have set the replicationFactor to different values for HA throughout the pipelines.

Reply:
Why do we keep the replication-factor in the WhiskConfig?

Reply:
as above.  Because @markusthoemmes was on vacation last week and I am not clear on the plan for how values that are very likely to be changed in different deployment need to be managed, I was leaving for followup PR.  I'm blocked on some other kube work (elimintating custom kafka image) until this PR gets merged.

Reply:
Is there a chance to move the replicationFactor from WhiskConfig to Pureconfig in this PR?

Reply:
Since Markus is back to answer questions, yeah I can do that in this PR too.